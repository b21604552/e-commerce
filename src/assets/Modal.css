.modal-wrapper {
    width: 400px;
    height: 260px;
    background: #FFFFFF;
    border: 1px solid rgba(228, 228, 228, 0.6);
    box-sizing: border-box;
    box-shadow: 0 6px 12px rgba(50, 50, 71, 0.07);
    border-radius: 4px;
    z-index: 10000000;
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
}

.modal-header {
    margin: 24px 38px 20px 16px;
    font-style: normal;
    font-weight: bold;
    font-size: 14px;
    line-height: 14px;
}

.modal-divider {
    border: 1px solid #CDCDCD;
    width: 100%;
}

.modal-body {
    margin: 16px 38px 10px 20px;
    width: 342px;
    height: 128px;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 17px;
    color: #484848;
}

.modal-footer {
    height: 26px;
    margin-right: 20px;
    display: inline-flex;
    float: right;
    justify-content: center;
}

.modal-button {
    width: 70px;
    height: 26px;
    border: 0;
    border-radius: 4px;
    font-style: normal;
    font-weight: bold;
    font-size: 12px;
    line-height: 12px;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    color: #FFFFFF;
    margin-left: 10px;
    cursor: pointer;
}

.success-button {
    background: #90D659;
}

.error-button {
    background: #D65959;
}