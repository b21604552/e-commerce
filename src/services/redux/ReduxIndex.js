import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
