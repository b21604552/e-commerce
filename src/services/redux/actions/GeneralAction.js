import { reduxStore } from '../ReduxIndex'

export function setHeaderTitle (headerTitle) {
  reduxStore.dispatch({
    type: 'setHeaderTitle',
    payload: {
      headerTitle
    },
  })
}

export function setSearchedKeyword (searchKeyword) {
  reduxStore.dispatch({
    type: 'setSearchedKeyword',
    payload: {
      searchKeyword
    },
  })
}

export function setOrderValue (orderValue) {
  reduxStore.dispatch({
    type: 'setOrderValue',
    payload: {
      orderValue
    },
  })
}

export function setItemCountInBasket (itemCountInBasket) {
  reduxStore.dispatch({
    type: 'setItemCountInBasket',
    payload: {
      itemCountInBasket
    },
  })
}

export function setModalFinished (modalFinished) {
  reduxStore.dispatch({
    type: 'setModalFinished',
    payload: {
      modalFinished
    },
  })
}

export function setModalVisible (modalVisible) {
  reduxStore.dispatch({
    type: 'setModalVisible',
    payload: {
      modalVisible
    },
  })
}
