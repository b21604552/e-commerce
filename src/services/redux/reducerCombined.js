import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
