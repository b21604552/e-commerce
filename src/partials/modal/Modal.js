import React from 'react'
import { setModalFinished, setModalVisible } from '../../services/redux/actions/GeneralAction'
import { reduxStore } from '../../services/redux/ReduxIndex'

export default function Modal () {
  const { modalVisible } = reduxStore.getState().GeneralReducer

  function modalFinished (value) {
    setModalFinished(value)
    setModalVisible(false)
  }

  return (
    <>
      {
        modalVisible && (
          <div className={'modal-wrapper'}>
            <div className={'modal-header'}>
              Ürünü silmek istediğinize emin misiniz?
            </div>
            <div className={'modal-divider'}/>
            <div className={'modal-body'}>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a
              galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentiall....
            </div>
            <div className={'modal-footer'}>
              <button id={'modal-confirm'} className={'modal-button success-button'} onClick={() => modalFinished(true)}>EVET</button>
              <button id={'modal-reject'} className={'modal-button error-button'} onClick={() => modalFinished(false)}>HAYIR</button>
            </div>
          </div>
        )
      }
    </>
  )
}