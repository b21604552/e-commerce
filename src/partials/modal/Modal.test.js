import React from 'react'
import ReactDOM from 'react-dom'
import Modal from './Modal'
import renderer from 'react-test-renderer'
import { reduxStore } from '../../services/redux/ReduxIndex'
import { setModalVisible } from '../../services/redux/actions/GeneralAction'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import { Provider } from 'react-redux'

configure({ adapter: new Adapter() })

describe('Modal Component Test', () => {
  const page = (
    <Provider store={reduxStore}>
      <Modal/>
    </Provider>
  )

  it('Renders Modal without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(page, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders Modal correctly', () => {
    const div = renderer.create(page).toJSON()
    expect(div).toMatchSnapshot()
  })

  it('Modal Change Test 1', () => {
    setModalVisible(true)
    const modal = shallow(page)
    modal.dive().find(Modal).dive().find('#modal-confirm').simulate('click')
    const { modalVisible } = reduxStore.getState().GeneralReducer
    expect(modalVisible).toStrictEqual(false)
  })

  it('Modal Change Test 2', () => {
    setModalVisible(true)
    const modal = shallow(page)
    modal.dive().find(Modal).dive().find('#modal-reject').simulate('click')
    const { modalVisible } = reduxStore.getState().GeneralReducer
    expect(modalVisible).toStrictEqual(false)
  })
})