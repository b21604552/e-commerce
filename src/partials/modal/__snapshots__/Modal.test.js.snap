// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Test Renders Modal correctly 1`] = `null`;
